pkg/volume/csi/csi_attacher.go

```diff
@@ -197,8 +201,14 @@ func (c *csiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.No
 		}
 
 		attachID := getAttachmentName(volumeHandle, driverName, string(nodeName))
+		attach, err := c.plugin.volumeAttachmentLister.Get(attachID)
+		if err == nil {
```

Comment:
How about changing this to `if err != nil ...` and moving the entire block of querying the API server inside that? That also unifies how `attached[spec]` is assigned and logging.

Reply:
did some refactoring here.